### ISSUE TITLE
¿cómo dockerizar staging de ckan?

### ISSUE THREAD

**Issue by rodowi** [mxabierto/devops#1]

**Comment by poguez** [comment]
Configuración actual con Postgres 8.4 y "ckan_version": "2.2.1b".
Está probado que funciona bien con postgres 9.1.

**Comment by poguez** [comment]
Primero sería mejor probar con los dumps generados por ckan como menciona [Upgrading CKAN- backup your database](http://docs.ckan.org/en/latest/maintaining/upgrading/index.html#backup-your-database) que nos menciona sobre los [comandos que de paster de dump y load de un backup](http://docs.ckan.org/en/latest/maintaining/paster.html#paster-db).

Otra posibilidad alternativa es con "pg_upgrade"
```"pg_upgrade" supports upgrades from 8.3.X and later to the current major release of PostgreSQL, including snapshot and alpha releases.```

Hay que tener en cuenta algunas configuraciones probadas para mantener queries geoespaciales posteriormente, si la imagen es con ubuntu 14.04 como base de acuerdo con  [Ckanext-spatial upgrade](http://docs.ckan.org/projects/ckanext-spatial/en/latest/install.html?highlight=upgrade).

**Comment by rodowi** [comment]
@poguez creo que habíamos dicho que no era necesario en estos momentos los queries geoespaciales, porque nunca usamos ese feature de búsqueda por bounding-box cc: @urkonn

**Comment by urkonn** [comment]
Me falló la importación de la base de datos, probablemente por las tablas geoespaciales

Usando como base el contenedor de PostgreSQL podemos tener PostGIS 

```
FROM postgres:9.4

RUN apt-get update && apt-get install -y --no-install-recommends postgresql-9.4-postgis libxml2-dev libxslt1-dev libgeos-c1
```

A partir del contenedor de CKAN podríamos instalar las extensiones que nos hagan falta 

```
FROM mxabierto/ckan

RUN apt-get install python-dev libxml2-dev libxslt1-dev libgeos-c1

RUN virtualenv $CKAN_HOME && \
  pip install -e "git+https://github.com/ckan/ckanext-spatial.git#egg=ckanext-spatial" && \
  pip install -r /src/ckanext-spatial/pip-requirements.txt
```

**Comment by poguez** [comment]
Yo lo pude lograr sin registrar en el archivo de configuracion `staging.ini` el `ckanext-spatial`. La información Geoespacial se podría regenerar posteriormente y así evitar la migración de Postgis inicialmente.

Por lo pronto pude restablecer bien el dump del comando paster.
La referencia de las tareas de base de datos de paster están en [link](http://docs.ckan.org/en/1117-start-new-test-suite/paster.html). Para generarlo lo hicimos desde el root de *ckan* con:
- `paster db dump -c /etc/ckan/default/production.ini my_database_dump.sql`

Posteriormente para reestablecerlo lo pude hacer exitoso sin registrar todos los plugins, especialmente no registré el espacial:
<img width="432" alt="screen shot 2015-12-08 at 5 01 36 pm" src="https://cloud.githubusercontent.com/assets/732015/11721763/14fb0e26-9f2b-11e5-9083-40349d6ddc97.png">

El reestablecimiento se logra:
- `paster db clean -c /etc/ckan/default/production.ini`
- `paster db load -c /etc/ckan/default/production.ini my_database_dump.sql`

Al final se debe de ver algo así:
<img width="1191" alt="screen shot 2015-12-08 at 4 58 18 pm" src="https://cloud.githubusercontent.com/assets/732015/11721864/8b50785e-9f2b-11e5-9963-84f15e4986a4.png">

**Comment by urkonn** [comment]
@poguez ¿eso ya es en contenedores?

**Comment by poguez** [comment]
No, para eso hay que hacer un contenedor/script para la carga del dump en la base de datos.

**Comment by urkonn** [comment]
@poguez precisamente

**Comment by urkonn** [comment]
Usé una máquina vagrant con las versiones de CKAN y PostgreSQL que vamos a usar para importar la base de datos, exporté nuevamente la base de datos de la máquina vagrant y la importé en el contenedor usando paster.

**Comment by urkonn** [comment]
## Lista de plugins que pueden ser útiles

- https://github.com/okfn/recline                                  Visualizaciones de datos tabulares
- https://github.com/ckan/ckanext-pages                     Creación de páginas tipo CMS
- https://github.com/ontodia/ckanext-discourse            Integración con un foro Discourse
- https://github.com/ckan/ckanext-disqus                    Sistema de comentarios con Disqus
- https://github.com/conwetlab/ckanext-datarequests   Permite a los usuarios solicitar datasets que no han sido publicados
- https://github.com/datacats/ckanext-featuredviews     Muestra las vistas de recursos que pueden tener visualizaciones o de buena calidad
- https://github.com/ckan/ckanext-qa                          Califica el nivel de apertura del dataset
- https://github.com/alantygel/ckanext-tagmanager      Ayuda a limpiar los tags
- https://github.com/ckan/ckanext-geoview                  Visualizaciones geoespaciales
- https://github.com/ckan/ckanext-deadoralive             Verifica que no existan ligas rotas
